atlasaction/action.go

```diff
@@ -747,6 +824,82 @@ func (a *Actions) SchemaApply(ctx context.Context) error {
 	return nil
 }
 
+// MonitorSchema runs the Action for "ariga/atlas-action/monitor/schema"
+func (a *Actions) MonitorSchema(ctx context.Context) error {
+	db, err := url.Parse(a.GetInput("url"))
+	if err != nil {
+		return err
+	}
+	var (
+		id = cloud.ScopeIdent{
+			URL:     db.Redacted(),
+			ExtID:   a.GetInput("slug"),
+			Schemas: nilIfEmpty(a.GetArrayInput("schemas")),
+			Exclude: nilIfEmpty(a.GetArrayInput("exclude")),
+		}
+	)
+	cc, err := a.CloudClient("cloud-token")
+	if err != nil {
+		return err
+	}
+	res, err := a.Atlas.SchemaInspect(ctx, &atlasexec.SchemaInspectParams{
+		URL:     a.GetInput("url"),
+		Schema:  id.Schemas,
+		Exclude: id.Schemas,
+		Format:  `{{ printf "# %s\n%s" .Hash .MarshalHCL }}`,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to inspect the schema: %w", err)
+	}
+	h, err := cc.SnapshotHash(ctx, &cloud.SnapshotHashInput{ScopeIdent: id})
+	if err != nil {
+		return fmt.Errorf("failed to get the schema snapshot hash: %w", err)
+	}
+	var (
+		parts = strings.SplitN(res, "\n", 2)
+		hash  = strings.TrimPrefix(parts[0], "# ")
+		hcl   = parts[1]
+		input = &cloud.PushSnapshotInput{
+			ScopeIdent: id,
+			HashMatch:  strings.HasPrefix(h, "h1:") && OldAgentHash(hcl) == h || hash == h,
+		}
+	)
+	if !input.HashMatch {
+		input.Snapshot = &cloud.SnapshotInput{Hash: hash, HCL: hcl}
+	}
+	u, err := cc.PushSnapshot(ctx, input)
+	if err != nil {
+		return fmt.Errorf("failed to push the schema snapshot: %w", err)
+	}
+	a.SetOutput("url", u)
+	a.Infof(`"atlas monitor schema" completed successfully, see the schema in Atlas Cloud: %s`, u)
+	return nil
+}
+
+func (a *Actions) CloudClient(tokenInput string) (CloudClient, error) {
+	t := a.GetInput(tokenInput)
+	if t == "" {
+		return nil, fmt.Errorf("missing required input: %q", tokenInput)
+	}
+	return a.cloudClient(t), nil
```

Comment:
The `cloudClient` call should able to return the error, if it can't create the new client. ie: invalid token was passed.
```suggestion
	return a.cloudClient(t)
```